src/upload-tests.proj

```diff
@@ -77,6 +77,12 @@
     <HelixArchLabel>$(ArchGroup)</HelixArchLabel>
     <HelixConfigLabel>$(ConfigurationGroup)</HelixConfigLabel>
     <MaxRetryCount>1</MaxRetryCount>
+    <!-- Unique container name per build and configuration, e.g., corefx-beta-12345-01-windows-x64 -->
+    <!-- Azure Storage containers can only contain [a-z, 0-9, -] in their names and $(OSGroup) is "Windows_NT" on Windows... -->
+    <ProcessedOSGroup>$(OSGroup.ToLower())</ProcessedOSGroup>
+    <ProcessedOSGroup Condition="'$(TargetsWindows)' == 'true'">windows</ProcessedOSGroup>
+    <ProcessedArchGroup>$(ArchGroup.ToLower())</ProcessedArchGroup>
+    <ContainerName Condition="'$(ContainerName)' == '' And '$(IsOfficial)' != ''">corefx-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(ProcessedOSGroup)-$(ProcessedArchGroup)</ContainerName>
```

Comment:
For corefx you also should take TargetGroup into consideration otherwise you might end up with duplicate names across legs.

Reply:
To see the list of target groups we support look in https://github.com/dotnet/corefx/blob/master/tools-local/targetgroups.props#L26. You can see a number of them have `.` in them is that a valid char for a container name?

Reply:
Good catch!  I had been running off the list [here](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md).  I'll add some processing to remove the `.`'s in those names.